godtools/App/Features/AppLanguage/Data/DownloadedLanguagesRepository/Cache/RealmDownloadedLanguage.swift

```diff
@@ -13,6 +13,7 @@ class RealmDownloadedLanguage: Object {
     
     @objc dynamic var createdAt: Date = Date()
     @objc dynamic var languageId: String = ""
+    @objc dynamic var downloadProgress: Double = 0
```

Comment:
Hey @rachaelblue what is your thinking here on persisting the download progress?  Would this be because if there's only a partial download and then something interrupts the download process?

Reply:
This was more of an easy and temporary solution to maintain the download animation before the real download logic gets implemented 😄 I expect we don't want this persisted, although I'm not sure what happens in the case of a partial download... would it just restart the whole thing?

Reply:
I was suspecting that it was temporary.  We could track the progress, although I was thinking we'd just restart the download.
But, this just made me think of something else.  I'm thinking that whenever a user starts a download of a language, we'd want to store / track that downloadable language immediately in the repo.
There is going to be some additional logic that checks which languages have been "downloaded" in that repo, and then download any latest translations that haven't been downloaded.
I'm thinking we'd store that downloadable language immediately to the repo in the event something does interrupt the app, then next app launch the language will get picked up in our process that downloads latest translations for those.

Reply:
Gotcha, so a flag or something that gets set on the Realm object once download completes?  I'm assuming that will happen in a later ticket?

Reply:
It wouldn't be an additional attribute, it would instead be a matter of when we store a language as "downloaded" in the DownloadedLanguagesRepository since we delete those when a language is removed.

In our case I'd say it would be safer to store a language as "downloaded" immediately when the translation downloading begins.  Especially now that we know those should continue downloading in the background.  This way if the download process is interrupted by the app getting closed etc, we can still download translations for the language next time the app is launched.

If we wait to store a language as downloaded when all translations have completed, we run the risk of losing that a language as marked for download if for some reason the download process is interrupted.

Reply:
That can be part of GT-2234 for translation downloads.